libcxx/include/__algorithm/min_element.h

```diff
@@ -27,20 +28,115 @@ _LIBCPP_PUSH_MACROS
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
+template <class _Iter, class _Sent, class = void>
+struct __ConstTimeDistance : false_type {};
```

Comment:
Let's rename this to `__has_constant_time_distance` and move it to `libcxx/include/__iterator/iterator_traits.h`. It can be useful elsewhere.